microbench/src/main/java/io/netty/microbench/internal/PrivilegedSocketOperationsBenchmark.java

```diff
@@ -0,0 +1,193 @@
+                        }
+                    });
+            ssc.close();
+            return ssc;
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithSMWithNullCheck(final SecurityManagerInstalled sm) throws IOException {
+        if (System.getSecurityManager() != null) {
+            try {
+                final ServerSocketChannel ssc = AccessController.doPrivileged(
+                        new PrivilegedExceptionAction<ServerSocketChannel>() {
+                            @Override
+                            public ServerSocketChannel run() throws Exception {
+                                final ServerSocketChannel ssc = ServerSocketChannel.open();
+                                ssc.socket().bind(null);
+                                ssc.configureBlocking(false);
+                                ssc.accept();
+                                return ssc;
+                            }
+                        });
+                ssc.close();
+                return ssc;
+            } catch (final PrivilegedActionException e) {
+                throw (IOException) e.getCause();
+            }
+        } else {
+            // this should never happen during benchmarking, but we write the correct code here
+            final ServerSocketChannel ssc = ServerSocketChannel.open();
+            ssc.socket().bind(null);
+            ssc.configureBlocking(false);
+            ssc.accept();
+            ssc.close();
+            return ssc;
+        }
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithoutSMNoPrivileged(final SecurityManagerEmpty sm) throws IOException {
+        final ServerSocketChannel ssc = ServerSocketChannel.open();
+        ssc.socket().bind(null);
+        ssc.configureBlocking(false);
+        ssc.accept();
+        ssc.close();
+        return ssc;
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithoutSM(final SecurityManagerEmpty sm) throws IOException {
+        try {
+            final ServerSocketChannel ssc = AccessController.doPrivileged(
+                    new PrivilegedExceptionAction<ServerSocketChannel>() {
+                        @Override
+                        public ServerSocketChannel run() throws Exception {
+                            final ServerSocketChannel ssc = ServerSocketChannel.open();
+                            ssc.socket().bind(null);
+                            ssc.configureBlocking(false);
+                            ssc.accept();
+                            return ssc;
+                        }
+                    });
+            ssc.close();
+            return ssc;
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithoutSMWithNullCheck(final SecurityManagerEmpty sm) throws IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "sm". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)